mirroroperator/registrymirror.py

```diff
@@ -459,6 +493,20 @@ def generate_stateful_set(self):
         stateful_set.spec.update_strategy = client.V1beta1StatefulSetUpdateStrategy(type="RollingUpdate",)
         return stateful_set
 
+    def generate_secret(self, secret):
+        secret.metadata = copy.deepcopy(self.metadata)
+        secret.metadata.name = self.nginx_config_secret_name
+        upstream_credentials = self.get_upstream_credentials()
+        auth = ''
+        if upstream_credentials:
+            (upstream_user, upstream_pass) = upstream_credentials
+            basic_auth = "{upstream_user}:{upstream_pass}".format(upstream_user=upstream_user, upstream_pass=upstream_pass)
```

Comment:
nit: 
possibly a little verbose, but it is clear.
this could probably be simplified to 
```
':'.join([upstream_user, upstream_pass])
```
/shrug

Reply:
In fact, join can accept a tuple too, so this could potentially be `':'.join(upstream_credentials)` ?

Reply:
up to you, whichever meets pep-20 best